pkg/logql/syntax/expr.y

```diff
@@ -272,13 +274,21 @@ filterOp:
   IP { $$ = OpFilterIP }
   ;
 
+orFilter:
```

Comment:
Hm, why is this required? Shouldn't it be enough to add it to line filter?

Reply:
The grammar for filter is: `filter STRING` which includes the filter token (something like `|=`). If we made and `OrFilter`

```
lineFilter:
    filter STRING                                                   { $$ = newLineFilterExpr($1, "", $2) }
  | filter filterOp OPEN_PARENTHESIS STRING CLOSE_PARENTHESIS       { $$ = newLineFilterExpr($1, $2, $4) }
  | filter STRING OR lineFilter                               { $$ = }
  ;
```

The corresponding syntax would be `|= something or |= something` which is clunky but also impossible because of the `lineFilters` grammar that creates a nested expression any time there are line filters:
```
lineFilters:
    lineFilter                { $$ = $1 }
  | lineFilters lineFilter    { $$ = newNestedLineFilterExpr($1, $2) }
  ;
```

If the grammar were, instead:
```
lineFilter:
    filter STRING                                                   { $$ = newLineFilterExpr($1, "", $2) }
  | filter filterOp OPEN_PARENTHESIS STRING CLOSE_PARENTHESIS       { $$ = newLineFilterExpr($1, $2, $4) }
  | filter STRING OR STRING                               { $$ = }
  ;
```
We would only be able to have one `or` expression: `|= "foo" or "bar"` but `|= "foo" or "bar" or "baz"` would be invalid

To get an arbitrary number of `or` statements that don't also require the `filter` element, we have to define our own recursive type that can then be added to the `lineFilter` grammar.